### PR TITLE
Fix whitespace bug in formulas parser

### PR DESCRIPTION
**Pull request by emhomm4**
Currently, the named formulas parser has a bug where it is not resilient to spaces in a formula. This PR fixes that. Unit tests updated to include random whitespace throughout formulas.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs [hunk 1]
```diff
@@ -655,25 +655,26 @@ public void TestParseTables_Negative(string script)
...
-        [InlineData("a=10;b")]
-        [InlineData("A=If(true,1;);")]
-        [InlineData("A=If(true,1;2);")]
-        [InlineData("1+2);")]
-        [InlineData("1=10")]
+        [InlineData("a = 10;;")]
+        [InlineData("a = 10; b")]
+        [InlineData("A = If(true,1;);")]
+        [InlineData("A = If(true,1;2);")]
+        [InlineData("1 + 2);")]
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs:676]: what about multi character names?
Multi character names with special characters?
Ones with whitespace in the name, surrounded by `'`?

Comment (emhomm4) [reply src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs:676]: Is there a way for me to know which special characters are allowed in identifiers?

Comment (emhomm4) [reply src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs:678]: I updated the tests. I don't think any special characters are allowed in identifiers, but correct me if I'm wrong.

Comment (emhomm4) [reply src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs:680]: Added tests with special characters inside of `'`

**Review by lesaltzm — APPROVED** [review]

**Review by VamsiModem — APPROVED** [review]

**Pull request merged**